Packs/SophosXGFirewall/Integrations/sophos_firewall.py

```diff
@@ -0,0 +1,2027 @@
+        raise Exception('Administrator type was selected. Please provide profile.')
+    json_data = {
+        'Username': username,
+        'Name': name,
+        'Password': password,
+        'UserType': user_type,
+        'Profile': profile,
+        'EmailList': {
+            'EmailID': email
+        },
+        'Group': group,
+        'Description': description,
+        'SurfingQuotaPolicy': surfing_quota_policy,
+        'AccessTimePolicy': access_time_policy,
+        'SSLVPNPolicy': ssl_vpn_policy,
+        'ClientlessPolicy': clientless_policy,
+        'DataTransferPolicy': data_transfer_policy,
+        'SimultaneousLoginsGlobal': simultaneous_logins_global,
+        'ScheduleForApplianceAccess': schedule_for_appliance_access,
+        'QoSPolicy': qos_policy,
+        'LoginRestriction': login_restriction,
+    }
+    return remove_empty_elements(json_data)
+
+
+def check_error_on_response(response: dict) -> None:
+    """Check if there is an error on the response
+
+    Args:
+        response (dict): the object to check the error on
+
+    Raises:
+        Exception: if there if an error in the response
+        Exception: if there are no records on list or get
+    """
+    response_message = retrieve_dict_item_recursively(response, '#text')
+    response_code = retrieve_dict_item_recursively(response, '@code')
+    response_status = retrieve_dict_item_recursively(response, 'Status')
+
+    if response_message and 'successful' not in response_message:
+        if response_code and int(response_code) > 299:
+            raise Exception(f'{response_message} (error code: {response_code})')
+    if response_status and 'No. of records Zero.' in response_status:
+        raise Exception(response_status)
+
+
+def generic_add(client: Client, constants: dict, params: dict, builder: Callable) -> CommandResults:
+    """Generic add function, sends false as to_update to generic save function
+
+    Args:
+        client (Client): Sophos XG Firewall Client
+        constants (dict): CONSTANT with the endpoint tag and table headers
+        params (dict): params for the builder
+        builder (Callable): the builder to build the object
+
+    Returns:
+        CommandResults: Command results object
+    """
+    to_update = False
+    return generic_save(client, params, builder, to_update, **constants)
+
+
+def generic_update(client: Client, constants: dict, params: dict,
+                   builder: Callable) -> CommandResults:
+    """Generic update function, sends true as to_update to generic save function
+
+    Args:
+        client (Client): Sophos XG Firewall Client
+        constants (dict): CONSTANT with the endpoint tag and table headers
+        params (dict): params for the builder
+        builder (Callable): the builder to build the object
+
+    Returns:
+        CommandResults: Command results object
+    """
+    to_update = True
+    return generic_save(client, params, builder, to_update, **constants)
```

Comment:
these two functions do not do the same action as generic_get/generic_save
they merely call them.
Have you considered removing them, and move the `to_update` argument in the signature of the function?